File: algorithms/strings/check_permutation/check_permutation.ts
function checkPermutation(s1: string, s2: string): boolean {
    if (s1.length != s2.length) return false;

    let chars: number[] = fill(256, 0);

    for (let i = 0; i < s1.length; i++)
        chars[s1.charCodeAt(i)]++;

    for (let i = 0; i < s2.length; i++) {
        chars[s2.charCodeAt(i)]--;
        if (chars[s2.charCodeAt(i)] < 0)
            return false;
    }

    return true;
}

function fill(len: number, el: number): number[] {
    const arr: number[] = Array(len);
    for (let i = 0; i < len; i++) arr[i] = el;
    return arr;
}

const wordsList = [["apple", "papel"], ["mango", "orange"], ["aaaa", "aaa"]];

for (const words of wordsList)
    console.log(`${words[0]} ${words[1]} : ${checkPermutation(words[0], words[1])}`);

File: algorithms/strings/check_permutation/check_permutation.rs
fn check_permutation(s1: &str,s2: &str) -> bool {
    if s1.len() != s2.len() {
        return false
    }

    let mut chars: [i32;256] = [0;256];

    for ch in s1.chars() {
        chars[ch as usize] += 1;
    }

    for ch in s2.chars() {
        chars[ch as usize] -= 1;
        if chars[ch as usize] < 0 {
            return false
        }
    }

    true
}

fn main() {

    // Probably a better way to do this
    let words: [String;6] = [
        String::from("apple"),
        String::from("papel"),
        String::from("mango"), 
        String::from("orange"), 
        String::from("aaaa"), 
        String::from("aaa")
    ];

    let mut i = 0;

    loop {
        if i >= words.len() {
            break;
        }
        
        println!("{} {} : {}", words[i], words[i+1], check_permutation(&words[i], &words[i+1]));

        i += 2;
    }

}